src/Microsoft.AspNet.Http.Abstractions/HttpResponse.cs

```diff
@@ -24,9 +25,9 @@ public abstract class HttpResponse
 
         public abstract bool HasStarted { get; }
 
-        public abstract void OnResponseStarting(Action<object> callback, object state);
+        public abstract void OnResponseStarting(Func<object, Task> callback, object state);
```

Comment:
Lou said no sync sugar that calls async and Wait. Here you'd just wrap the Action in a Func and return a completed Task.

That said, does it confuse the intellisense for anonymous delegates if you have overloads that differ only by return value?